### PR TITLE
[feature] optional auto_create setting to improve login UX

### PR DESCRIPTION
**Pull request by luiyongsheng**
This change/enhancement is based on UX optimisation consideration.

**Current Issue**
First-time guest visiting guarded pages will be redirected to login page.
Clicking social media icons at login page will return an error, which is counter-intuitive from UX perspective.

**Enhancement**
First-time guest can quickly get onboard when they visited links that required authentication.

### CODE REVIEW

**Comment by luiyongsheng** [comment]
This PR will make these kind of frontend design achievable
image

**Comment by srdante** [comment]
I think that would be something interesting as an optional feature

**Comment by joelbutcher** [comment]
Why not just change the destination of your auth guard?

**Comment by luiyongsheng** [comment]
> Why not just change the destination of your auth guard?

Technically we are not able to accurately predicts whether a guest is actually first-time visitor.

I don't think spending resources and effort from the backend deciding whether to present user with a login page or sign up page is efficient in terms of UX.

Plus, I already receive real customer feedback regarding this issue.

Really hope this PR is able to merge.

Thank you in advance.

**Comment by srdante** [comment]
I think this should be enabled or disabled through a configuration parameter. Many people prefer the warning that there is no account registered with this data rather than creating it automatically.

**Comment by luiyongsheng** [comment]
https://github.com/joelbutcher/socialstream/pull/49/commits/d95b761832c1c664d2f5ead988816b3ac4885e43 tested in sandbox/staging servers and worked properly.

File: src/Http/Controllers/OAuthController.php [hunk 1]
```diff
@@ -124,8 +124,10 @@ public function handleProviderCallback(Request $request, string $provider)
             ]);
         }
 
+        $is_autocreate = config('socialstream.auto_create');
```
Comment (joelbutcher) [inline src/Http/Controllers/OAuthController.php:127]: Please use camelCase for variable names

**Review by joelbutcher — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/OAuthController.php (hunk 1 above)
Comment (luiyongsheng) [reply src/Http/Controllers/OAuthController.php]: Changes made https://github.com/joelbutcher/socialstream/pull/49/commits/a1cea6eb550b52f3102dca5ab61b0446b675436e

**Comment by luiyongsheng** [comment]
Changes made https://github.com/joelbutcher/socialstream/commit/a1cea6eb550b52f3102dca5ab61b0446b675436e

**Comment by luiyongsheng** [comment]
Is this going to merge?

**Comment by joelbutcher** [comment]
@luiyongsheng I'd like to follow Jetstream's example in keeping a "features" array in the config. Can you create a PR that adds this as `Socialstream::registerOnLogin()`?

**Pull request closed without merging**